### ISSUE TITLE
postgresql prepared statement does not exist

### ISSUE THREAD

**Issue by raraodd** [processone/ejabberd#1325]
> What version of ejabberd are you using?

16.09

> What operating system (version) are you using?

Centos 6.8

> How did you install ejabberd (source, package, distribution)?

source

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I am using ejabberd with postgre as login auth. I got an error after I restarted pgsql server and tried to login.

```
2016-10-07 16:09:27.656 [error] <0.863.0>@ejabberd_sql:check_error:1033
 SQL query 'Q9525209' at {sql_queries,145}
 failed: [{severity,'ERROR'},{code,<<"26000">>},
  {message,<<"prepared statement \"Q9525209\" does not exist">>},
  {file,<<"prepare.c">>},{line,505},{routine,<<"FetchPreparedStatement">>}]
```

Scenario:
1. Start postgre
2. Start ejabberd
3. User login
4. User logout
5. Restart postgre
6. User login but failed

SQL config:

```
sql_type: pgsql
sql_server: "db"
sql_port: 5432
sql_username: "cat"
sql_password: "cat123"
sql_database: "catty"
sql_keepalive_interval: 600
auth_method: [sql]
```

**Comment by johnhamelink** [comment]
I'm also experiencing this issue - restarting the service fixed it for me, but I'm not sure why it happens in the first place?

**Comment by alexeyshch** [comment]
Fixed in f45dc468560fc9d4399e12e65c0ee3d6e10c53ae

**Comment by alok102singh** [comment]
@alexeyshch Hey i know this thread get closed. But i found a similar issue that's why i commented here.

``` 2017-05-01 16:06:04.219 [error] <0.8096.5>@ejabberd_sql:check_error:1033 SQL query 'Q9525209' at {sql_queries,145} failed: <<"internal error">>
2017-05-01 16:06:17.217 [error] <0.8113.5>@ejabberd_sql:check_error:1033 SQL query 'Q9525209' at {sql_queries,145} failed: [{severity,'ERROR'},{code,<<"26000">>},{message,<<"prepared statement \"Q9525209\" does not exist">>},{file,<<"prepare.c">>},{line,505},{routine,<<"FetchPreparedStatement">>}]
2017-05-01 16:06:28.255 [error] <0.8129.5>@ejabberd_sql:check_error:1033 SQL query 'Q9525209' at {sql_queries,145} failed: [{severity,'ERROR'},{code,<<"26000">>},{message,<<"prepared statement \"Q9525209\" does not exist">>},{file,<<"prepare.c">>},{line,505},{routine,<<"FetchPreparedStatement">>}]```

above this error i got following errors:
```2017-05-01 16:06:04.218 [error] <0.5982.0> gen_server <0.5982.0> terminated with reason: tcp_close
2017-05-01 16:06:04.218 [error] <0.5982.0> CRASH REPORT Process <0.5982.0> with 1 neighbours exited with reason: tcp_close in gen_server:terminate/7 line 812```

But my ejabberd version is 16.06
